default.nix

```diff
@@ -29,17 +29,21 @@
 
 , withHoogle  ? true
 
-, rev ? "29d57de30101b51b016310ee51c2c4ec762f88db" #  2020-05-23: NOTE: UTC 17:00
 
-, sha256 ? "1wjljkffb3gzdvpfc4v98mrhzack6k9i7860n8cf5nipyab6jbq9"
+, useRev ? false
+# Accepts Nixpkgs channel name and Git revision
+, rev ? "nixpkgs-unstable"
```

Comment:
Do I understand this correctly, that by default, `nix-build` or `nix-shell` will now use the latest `nixpkgs-unstable` version, which might be broken?

So could it happen, that a new contributor tries to use `nix-shell` but it doesn't currently work? I think that could scare off new contributors.

If it's important to _test_ `hnix` with `nixpkgs-unstable`, I'd rather have an additional Travis job that does this.

Reply:
That is a separate discussion.
`nixpkgs-unstable` is actually a default "stable" branch for Nix. It is not a `master` branch.

Due to purely functional nature, Nix packages are indeed brittle and can cascade break quite frequently. Moreover that Haskell people still not pay attention to Nixpkgs, despite Cabal using Nix internally and migration is in process. But those breakages should not happen in `nixpkgs-unstable` - that is their maintenance responsibility. We need to see what channel we must use, but completely freeze the project ecosystem is not an answer here. Before I updated `rev` in the previous PR, we were actually using 2 year old version on Nixpkgs.

I would say - yes - I agree - we need to be sure that HNix is buildable by other people. And we did not know how they build or develop it.
That is why we must make sure we develop it and it tests on current Hackage & Nixpkgs ecosystem. That is the only way to be sure it is really proper citizen in Haskell and Nixpkgs ecosystems. There is no way of stopping the progress, so we must follow it. Like we must follow `GHC` now into `8.10`, but frozen `rev`s probably had no `GHC 8.10`.

That is all a separate discussion.